tests/components/freebox/test_sensor.py

```diff
@@ -0,0 +1,124 @@
+"""Tests for the Freebox sensors."""
+from copy import deepcopy
+from datetime import timedelta
+from unittest.mock import Mock
+
+from homeassistant.components.freebox.const import DOMAIN
+from homeassistant.const import CONF_HOST, CONF_PORT
+from homeassistant.core import HomeAssistant
+from homeassistant.setup import async_setup_component
+import homeassistant.util.dt as dt_util
+
+from .const import (
+    DATA_CONNECTION_GET_STATUS,
+    DATA_HOME_GET_NODES,
+    DATA_STORAGE_GET_DISKS,
+    MOCK_HOST,
+    MOCK_PORT,
+)
+
+from tests.common import MockConfigEntry, async_fire_time_changed
+
+
+async def test_network_speed(hass: HomeAssistant, router: Mock) -> None:
+    """Test missed call sensor."""
+    entry = MockConfigEntry(
+        domain=DOMAIN,
+        data={CONF_HOST: MOCK_HOST, CONF_PORT: MOCK_PORT},
+        unique_id=MOCK_HOST,
+    )
+    entry.add_to_hass(hass)
+    assert await async_setup_component(hass, DOMAIN, {})
+    await hass.async_block_till_done()
+
+    assert hass.states.get("sensor.freebox_download_speed").state == "198.9"
+    assert hass.states.get("sensor.freebox_upload_speed").state == "1440.0"
+
+    # Simulate a changed speed
+    data_connection_get_status_changed = deepcopy(DATA_CONNECTION_GET_STATUS)
+    data_connection_get_status_changed["rate_down"] = 123400
+    data_connection_get_status_changed["rate_up"] = 432100
+    router().connection.get_status.return_value = data_connection_get_status_changed
+    # Simulate an update
+    async_fire_time_changed(hass, dt_util.utcnow() + timedelta(seconds=60))
```

Comment:
Last week, a lot of time based tests were replaced with freezegun, please check it out at other integrations

Reply:
freezegun: done